core/src/main/java/jenkins/agents/WebSocketAgents.java

```diff
@@ -145,10 +145,10 @@ protected void opened() {
         }
 
         @Override
-        protected void binary(byte[] payload, int offset, int len) {
-            LOGGER.finest(() -> "reading block of length " + len + " from " + agent);
+        protected void binary(ByteBuffer data) {
+            LOGGER.finest(() -> "reading block of length " + data.remaining() + " from " + agent);
             try {
-                transport.receive(ByteBuffer.wrap(payload, offset, len));
+                transport.receive(data);
```

Comment:
Remoting supports `ByteBuffer` natively, so we can pass Jetty's `ByteBuffer` directly to Remoting without additional wrapping here.